internal/alloy/componenttest/componenttest.go

```diff
@@ -163,6 +164,8 @@ func (c *Controller) buildComponent(dataPath string, args component.Arguments) (
 			switch name {
 			case labelstore.ServiceName:
 				return labelstore.New(nil, prometheus.DefaultRegisterer), nil
+			case livedebugging.ServiceName:
+				return livedebugging.NewDebugStreamManager(), nil
```

Comment:
Here we use just `New()` function: https://github.com/grafana/alloy/pull/797/files#diff-7e74a62873427a2bfee5c9d25f0e133d65d417d54910964d843e4a8f616825d1R276

Why is it NewDebugStreamManager here?

Reply:
Because when you call Data() on the labelstore service it returns a pointer to itself:
func (s *service) Data() any {
	return s
}
whereas the livedebugging service returns only the debugStreamManager